### PR TITLE
[nfc][ctx_prof] Factor the callsite instrumentation exclusion criteria

### PR DESCRIPTION
**Pull request by mtrofin**
Reusing this in the logic fetching the instrumentation in `CtxProfAnalysis`

### CODE REVIEW

File: llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll [hunk 1]
```diff
@@ -250,6 +250,25 @@ define void @no_counters() {
   call void @bar()
   ret void
 }
+
+define void @inlineasm() {
```
Comment (teresajohnson) [inline llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll:254]: I'm not sure what specifically we are looking for below with regards to the check affected by this patch. Maybe add a comment.

File: llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp [hunk 2]
```diff
@@ -945,9 +945,7 @@ void FunctionInstrumenter::instrument() {
       for (auto &BB : F)
         for (auto &Instr : BB)
           if (auto *CS = dyn_cast<CallBase>(&Instr)) {
-            if ((CS->getCalledFunction() &&
-                 CS->getCalledFunction()->isIntrinsic()) ||
-                dyn_cast<InlineAsm>(CS->getCalledOperand()))
+            if (!InstrProfCallsite::canInstrumentCallsite(*CS))
```
Comment (teresajohnson) [inline llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp:948]: Is this an NFC change? If so that could be noted in the title

**Comment by teresajohnson** [comment]
> Reusing this in the logic fetching the instrumentation in `CtxProfAnalysis`

Do you mean it will be reused in a future patch? Because I only see the one use here.

**Comment by mtrofin** [comment]
> > Reusing this in the logic fetching the instrumentation in `CtxProfAnalysis`
> 
> Do you mean it will be reused in a future patch? Because I only see the one use here.

yes

File: llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll (hunk 1 above)
Comment (mtrofin) [reply llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll:254]: done.

File: llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp (hunk 2 above)
Comment (mtrofin) [reply llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp:948]: done

File: llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll (hunk 1 above)
Comment (teresajohnson) [reply llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll:255]: But isn't the "llvm.instrprof.increment" instrumentation? Oh I see we are looking to make sure there is no "llvm.instrprof.callsite" instrumentation, right? Maybe just mention that.

**Review by teresajohnson — APPROVED** [review]

File: llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll (hunk 1 above)
Comment (mtrofin) [reply llvm/test/Transforms/PGOProfile/ctx-instrumentation.ll:255]: Yup - and clarified.

**Pull request merged**